accepted/0000-programming-style-guide.md

```diff
@@ -0,0 +1,103 @@
+# Paketo Project Programming Style Guide
+
+## Summary
+
+This RFC proposes a Programming Style Guide for the Paketo project. It will be
+used to evaluate and polish pull requests, educate contributors, and overall
+maintain consistent code quality in the project
+
+## Motivation
+
+The Programming Style Guide will addrress the challenge of maintaining
+consistent and high-quality code with excellent test coverage in an open source
+project. Explicitly naming expectations will benefit both contributors of new
+code and reviewers of that code in the following ways:
+- Contributors (and especially first-time contributors) will have a standard
+  against which to evaluate their work before opening it to outside feedback.
+  They can address problems before reviewers see them.
+- Contributors can refer to the Style Guide during ideation to understand which
+  of their ideas are likely to a) be useful and b) play well with existing
+  elements of the project.
+- Reviewers can ground their feedback by citing a centralized Style Guide that
+  may include more complete explanations of common feedback.
+- Reviewers can be confident that their feedback covers the most important
+  bases (i.e. those outlined in the Style Guide)
+
+Maintaining a Style Guide also improves the fairness and inclusivity of the
+project. Reviewers and contributors can ground their efforts in a set of
+explicit and agreed upon standards, which reduces the barrier to entry for new
+contributors to the project. Furthermore, standards-based feedback reduces the
+amount that implicit biases about whose work is "high quality" come into play.
+
+
+{{Why are we doing this? What pain points does this resolve? What use cases
+does it support? What is the expected outcome? Use real, concrete examples to
+make your case!}}
+
+## Detailed Explanation
+
+Explain the logistics of where the guide will be placed and referred to. Also
+talk about guiding principles for what's included in the guide.
```

Comment:
Is this a note to your future self that this section needs to be expanded?

Reply:
Yes - I'm in general trying to go about this RFC a bit differently than others. It seems like the topics in this note should be addressed after the content of a style guide has been mostly established.

Though I'm beginning to wonder if this RFC should only be about approving the *existence* of a programming guide, and the actual implementation of the guide can happen... via other RFCs? Or by some other mechanism? What do you think about that @ryanmoran?

Reply:
This sounds reasonable. I don't think it makes sense for us to try to develop the entirety of the styleguide in this RFC. It makes more sense to decide where it will reside and what its initial contents might be. I'd be happy to ✅ an RFC that answered those questions.